rest/src/main/java/org/jboss/pnc/rest/endpoints/internal/BuildTaskEndpointImpl.java

```diff
@@ -113,8 +167,194 @@ public Response buildTaskCompleted(String buildId, BuildResultRest buildResult)
     }
 
     @Override
+    @Deprecated // TODO remove once fully migrated to Rex
     public Response buildTaskCompletedJson(String buildId, BuildResultRest buildResult)
             throws org.jboss.pnc.facade.validation.InvalidEntityException {
         return buildTaskCompleted(buildId, buildResult);
     }
+
+    @Override
+    @Transactional
+    public Response buildTaskNotification(@NotBlank String buildId, @NotNull NotificationRequest notification)
+            throws InvalidEntityException {
+        logger.debug("Received transition notification for Build '{}'", buildId);
+
+        if (!buildId.equals(notification.getTask().getName())) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity("Build ID does not correspond to associated Rex Task")
+                    .build();
+        }
+
+        BuildMeta buildMeta = parseBuildMeta(buildId, notification);
+
+        // Task has to be present in Rex, otherwise it is unknown
+        validateBuildIsActive(buildId, notification);
```

Comment:
Isn't Rex who is invoking this endpoint? Why asking Rex again ?

Reply:
Because, without this check, you would be able to use this endpoint to create or send notification about a build record that may have never been scheduled by Orch. This could be potentially very harmful.

Reply:
This should be prevented by the authorization. Currently we are using user's token though, as we trust the users that they are not trying to break their own jobs. If required we might consider using Rex service account.